src/coreclr/vm/peassembly.h

```diff
@@ -78,19 +77,22 @@ typedef VPTR(PEAssembly) PTR_PEAssembly;
 //    for reflection-based modules.
 //
 // See also file:..\inc\corhdr.h#ManagedHeader for more on the format of managed images.
-// See code:Module for more on modules
 // --------------------------------------------------------------------------------
 
-typedef VPTR(class PEFile) PTR_PEFile;
-
-typedef ReleaseHolder<IMDInternalImport> IMDInternalImportHolder;
-
-class PEFile
+class PEAssembly
```

Comment:
Yes, it can be final. I think.
It needs to be virtual though. I could not convince DAC to build if the class does not have at least a virtual destructor.